src/components/PeoplePage.tsx

```diff
@@ -0,0 +1,83 @@
+import { useEffect, useState } from 'react';
+import { Loader } from './Loader';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { PersonLink } from './PersonLink';
+
+enum RenderStatus {
+  Loading = 'Loading',
+  Error = 'Error',
+  ShowPeople = 'ShowPeople',
+}
+
+export const PeoplePage = () => {
+  const [renderStatus, setRenderStatus]
+    = useState<RenderStatus>(RenderStatus.Loading);
+  const [people, setPeople] = useState<Person[] | null>(null);
+
+  const renderPeople = () => {
```

Comment:
interesting solution. You can wrap this render function with useCallback with those dependencies you need